### PR TITLE
Sonarjava 339 suppress warnings

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

**Comment by m-g-sonar** [comment]
Fix Windows 8 build:
- Force EOL to LF for sonar symbol table test file using .gitattributes
Added new rule:
- SONARJAVA-339 / RSPEC-1309: Added rule to manage warnings annotation. Allows to specify a single, or a list of warnings which are not allowed. By default, no '@SuppressWarning' will be allowed (empty list).

**Comment by benzonico** [comment]
See my comments on files.

**Comment by m-g-sonar** [comment]
Source code have been updated with respect to previous comments.

File: java-checks/src/main/java/org/sonar/java/checks/SuppressWarningsCheck.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      for (ExpressionTree expressionTree : array.initializers()) {
+        if (expressionTree.is(Kind.STRING_LITERAL)) {
+          result.add(getCleanedLitteralValue((LiteralTree) expressionTree));
+        }
+      }
+    }
+    return result;
+  }
+
+  private String getCleanedLitteralValue(LiteralTree tree) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/SuppressWarningsCheck.java]: Typo

File: java-checks/src/test/java/org/sonar/java/checks/SuppressWarningsCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  @Test
+  public void two_warnings_from_same_line_are_not_allowed() throws Exception {
+    CheckMessagesVerifier.verify(getTestFile("unchecked, cast").getCheckMessages())
+      .next().atLine(10).withMessage("Suppressing the 'unchecked', 'cast' warnings is not allowed")
+      .noMore();
+  }
+
+  private SourceFile getTestFile(String listOfWarnings) {
```
Comment (benzonico) [inline java-checks/src/test/java/org/sonar/java/checks/SuppressWarningsCheckTest.java]: getSourceFile ?

File: java-checks/src/test/files/checks/SuppressWarningsCheck.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+@SuppressWarnings("unused")
+class A {
+  
+  private final String BOXING = "boxing";
+  
+  @SuppressWarnings("all")
+  private  void f() {
+  }
```
Comment (benzonico) [inline java-checks/src/test/files/checks/SuppressWarningsCheck.java]: Spacing.

File: java-checks/src/test/files/checks/SuppressWarningsCheck.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+@SuppressWarnings("unused")
+class A {
+  
+  private final String BOXING = "boxing";
+  
+  @SuppressWarnings("all")
+  private  void f() {
```
Comment (benzonico) [inline java-checks/src/test/files/checks/SuppressWarningsCheck.java]: spacing

File: java-checks/src/main/java/org/sonar/java/checks/SuppressWarningsCheck.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        }
+      }
+    }
+  }
+
+  private boolean isJavaLangSuppressWarnings(AnnotationTree tree) {
+    return ((AbstractTypedTree) tree).getSymbolType().is("java.lang.SuppressWarnings");
+  }
+
+  private List<String> getRuleWarnings() {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/SuppressWarningsCheck.java]: Renaming method to forbiddenWarnings ?

**Comment by benzonico** [comment]
LGTM

**Pull request merged**